frontend/src/components/auth/Login.tsx

```diff
@@ -6,57 +6,73 @@ import AuthContext from "../../contexts/AuthContext";
 import { AuthenticatedUser } from "../../types/AuthTypes";
 import AUTHENTICATED_USER_KEY from "../../constants/AuthConstants";
 import { capitalizeFirstLetter } from "../../utils/StringUtils";
+import { authErrors } from "../../errors/AuthErrors";
+import { PresentableError } from "../../types/ErrorTypes";
+import { isRole } from "../../types/UserTypes";
 
 const Login = (): React.ReactElement => {
   const { authenticatedUser, setAuthenticatedUser } = useContext(AuthContext);
+  const [loginError, setLoginError] = useState<PresentableError>();
+  const [emailError, setEmailError] = useState<PresentableError>();
   const [email, setEmail] = useState("");
   const [password, setPassword] = useState("");
   const history = useHistory();
 
   const location = useLocation();
   const searchParams = new URLSearchParams(location.search);
-  const role = searchParams.get("role");
+  const role = capitalizeFirstLetter(searchParams.get("role"));
 
   if (authenticatedUser) {
     return <Redirect to={HOME_PAGE} />;
   }
 
-  if (!role || !["administrator", "facilitator", "learner"].includes(role)) {
+  if (!role || !isRole(role)) {
     return <Redirect to={WELCOME_PAGE} />;
   }
 
   const onLogInClick = async () => {
-    const user: AuthenticatedUser | null = await authAPIClient.login(
-      email,
-      password,
-    );
-    const isUserVerified = user?.accessToken
-      ? await authAPIClient.isUserVerified(email, user.accessToken)
-      : null;
-    if (!user || !isUserVerified) {
-      // will need to change this for different errors
-      // eslint-disable-next-line no-alert
-      alert("Bad login, user not found");
-      return;
-    }
-    if (user.role.toLowerCase() !== role.toLowerCase()) {
-      // change this later to not use an alert
-      // eslint-disable-next-line no-alert
-      alert(`Bad login. Expected ${user.role}, got ${role}`);
-      return;
-    }
+    try {
+      setLoginError(undefined);
+      setEmailError(undefined);
+      const user: AuthenticatedUser | null = await authAPIClient.login(
+        email,
+        password,
+        role,
+      );
 
-    localStorage.setItem(AUTHENTICATED_USER_KEY, JSON.stringify(user));
-    setAuthenticatedUser(user);
+      localStorage.setItem(AUTHENTICATED_USER_KEY, JSON.stringify(user));
+      setAuthenticatedUser(user);
+    } catch (e: unknown) {
+      if (e instanceof Error && e.message in authErrors) {
+        // eslint-disable-next-line no-alert
+        alert(e.message);
+      } else {
+        // eslint-disable-next-line no-alert
+        alert("bad!");
+      }
+    }
   };
 
   const onSignupClick = () => {
-    history.push(`${SIGNUP_PAGE}?role=${role}`);
+    history.push(`${SIGNUP_PAGE}?role=${role.toLowerCase()}`);
   };
 
   return (
     <div style={{ textAlign: "center" }}>
       <h1>{capitalizeFirstLetter(role)} Login</h1>
+      {loginError && (
```

Comment:
just note that this doesnt appear in the UI since it isn't be set in onLoginClick

Reply:
yupp no worries there, the rest of the frontend for the error handling will come after the login pages are fully implemented 🫡